runtime/src/chpl-launcher-common.c

```diff
@@ -636,12 +635,10 @@ void chpl_compute_real_binary_name(const char* argv0) {
   if (launcher_is_mli) {
 
     length = strlen(launcher_mli_real_name);
+    chpl_real_binary_name = chpl_mem_alloc(length+1,
+                                           CHPL_RT_MD_COMMAND_BUFFER, -1, 0);
 
-    if (length >= BIN_NAME_SIZE) {
-      chpl_internal_error("Real executable name is too long.");
-    }
-
-    strncpy(chpl_real_binary_name, launcher_mli_real_name, BIN_NAME_SIZE);
+    strncpy(chpl_real_binary_name, launcher_mli_real_name, length);
```

Comment:
I'm seeing build warnings for this like:

```
chpl-launcher-common.c: In function 'chpl_compute_real_binary_name':
chpl-launcher-common.c:641:5: warning: 'strncpy' specified bound depends on the length of the source argument [-Wstringop-overflow=]
  641 |     strncpy(chpl_real_binary_name, launcher_mli_real_name, length);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
chpl-launcher-common.c:637:14: note: length computed here
  637 |     length = strlen(launcher_mli_real_name);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
***** ./ *****
```

Where basically you're computing the length of the source instead of the dst, so this would still overflow if the source was larger than the dest